### PR TITLE
Feature/696

### PR DESCRIPTION
**Pull request by basmot**
Inform the ticket you are solving in this pull request: #696

### CODE REVIEW

File: exam_enrollment/locale/en/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -28,6 +28,9 @@ msgstr "Choose which exam you want to take for the next session exam."
 msgid "exams_enrollment"
 msgstr "Exams enrollment"
 
+msgid "exam_enrollment_form_submitted"
+msgstr "Exam enrollment form successfully submitted."
+
 msgid "info_exam_enrollment_by_default"
 msgstr "<b>By default, you are already enrolled to all checked exams below. "
        "<br>If it fit to your needs, you don't have any action to perform. </b>"
```
Comment (htmfilho) [inline exam_enrollment/locale/en/LC_MESSAGES/django.po]: if it fits your needs, then you don't have to do anything.

File: exam_enrollment/templates/exam_enrollment_form.html [hunk 2]
```diff
@@ -74,70 +74,76 @@
...
-                        <th>{% trans 'title' %}</th>
-                        <th>{% trans 'ects' %}</th>
-                        <th>{% trans 'inscr' %}</th>
-                        <th {% if current_number_session > 1 %} colspan="2" {% endif %} class="text-nowrap">{% trans 'jan' %} *</th>
-                        <th {% if current_number_session > 2 %} colspan="2" {% endif %} class="text-nowrap">{% trans 'june' %} *</th>
-                        <th class="text-nowrap">{% trans 'sept' %} *</th>
-                        <th>{% trans 'credited' %}</th>
-                    </tr>
-                    {% for exam_enrol in exam_enrollments %}
+                <form id="exam_enrollment_form" method="POST" action="#">
```
Comment (htmfilho) [inline exam_enrollment/templates/exam_enrollment_form.html]: L'action doit être différent de "#"

File: exam_enrollment/tests/ressources/exam_enrollment_form_example.json [hunk 3]
```diff
@@ -17,6 +17,7 @@
       "credits": 3.0,
       "credited": null,
       "enrolled_by_default": true,
+      "etat_to_inscr_current_session": "I",
```
Comment (htmfilho) [inline exam_enrollment/tests/ressources/exam_enrollment_form_example.json]: etat -> state

File: exam_enrollment/tests/ressources/exam_enrollment_form_example.json [hunk 4]
```diff
@@ -33,13 +34,13 @@
...
         "enrollment_state": "Insc."
-      },
-      "": ""
+      }
     },
     {
       "credits": 8.0,
       "credited": null,
       "enrolled_by_default": false,
+      "etat_to_inscr_current_session": "Y",
```
Comment (htmfilho) [inline exam_enrollment/tests/ressources/exam_enrollment_form_example.json]: etat -> state

File: exam_enrollment/tests/ressources/exam_enrollment_form_example.json [hunk 5]
```diff
@@ -62,6 +63,7 @@
       "credits": 10.0,
       "credited": "Report",
       "enrolled_by_default": true,
+      "etat_to_inscr_current_session": "",
```
Comment (htmfilho) [inline exam_enrollment/tests/ressources/exam_enrollment_form_example.json]: etat -> state

File: exam_enrollment/tests/ressources/exam_enrollment_form_example.json [hunk 6]
```diff
@@ -76,6 +78,7 @@
       "credits": 5.0,
       "credited": null,
       "enrolled_by_default": true,
+      "etat_to_inscr_current_session": "P",
```
Comment (htmfilho) [inline exam_enrollment/tests/ressources/exam_enrollment_form_example.json]: etat -> state

File: exam_enrollment/tests/ressources/exam_enrollment_form_example.json [hunk 7]
```diff
@@ -94,6 +97,7 @@
       "credits": null,
       "credited": null,
       "enrolled_by_default": false,
+      "etat_to_inscr_current_session": null,
```
Comment (htmfilho) [inline exam_enrollment/tests/ressources/exam_enrollment_form_example.json]: etat -> state

File: exam_enrollment/tests/views/test_exam_enrollment_form.py [hunk 8]
```diff
@@ -54,5 +81,69 @@ def test_form_is_correct(self):
...
 
+    @patch("osis_common.queue.queue_sender.send_message")
+    def test_exam_enrollment_form_submission_message(self, send_message):
+        send_message.return_value = None
+        self.client.force_login(self.user)
+        post_data = {
+            "chckbox_exam_enrol_sess1_LPHYS1234": "on",
+            "etat_to_inscr_current_session_LPHYS1234": "I",
+            "chckbox_exam_enrol_sess1_LBIO4567": "",
+            "etat_to_inscr_current_session_LBIO4567": "None",
```
Comment (htmfilho) [inline exam_enrollment/tests/views/test_exam_enrollment_form.py]: etat -> state

File: exam_enrollment/tests/views/test_exam_enrollment_form.py [hunk 9]
```diff
@@ -54,5 +81,69 @@ def test_form_is_correct(self):
...
+        response = self.client.get(self.url, follow=True)
+        self.assertTemplateUsed(response, 'access_denied.html')
 
+    @patch("osis_common.queue.queue_sender.send_message")
+    def test_exam_enrollment_form_submission_message(self, send_message):
+        send_message.return_value = None
+        self.client.force_login(self.user)
+        post_data = {
+            "chckbox_exam_enrol_sess1_LPHYS1234": "on",
+            "etat_to_inscr_current_session_LPHYS1234": "I",
```
Comment (htmfilho) [inline exam_enrollment/tests/views/test_exam_enrollment_form.py]: etat -> state

File: exam_enrollment/tests/views/test_exam_enrollment_form.py [hunk 10]
```diff
@@ -54,5 +81,69 @@ def test_form_is_correct(self):
...
+            "chckbox_exam_enrol_sess1_LBIO4567": "",
+            "etat_to_inscr_current_session_LBIO4567": "None",
+            "chckbox_exam_enrol_sess1_LDROI1111": None,
+            "etat_to_inscr_current_session_LDROI1111": None,
```
Comment (htmfilho) [inline exam_enrollment/tests/views/test_exam_enrollment_form.py]: etat -> state

File: exam_enrollment/tests/views/test_exam_enrollment_form.py [hunk 11]
```diff
@@ -54,5 +81,69 @@ def test_form_is_correct(self):
...
+            "current_number_session": 1,
+        }
+        # from django.contrib.messages.storage import default_storage
+        # request._messages = default_storage(request)
+        exam_enrollments_expected = [{"acronym": "LPHYS1234",
+                                      "is_enrolled": True,
+                                      "etat_to_inscr": "I"},
```
Comment (htmfilho) [inline exam_enrollment/tests/views/test_exam_enrollment_form.py]: etat -> state

File: exam_enrollment/tests/views/test_exam_enrollment_form.py [hunk 12]
```diff
@@ -54,5 +81,69 @@ def test_form_is_correct(self):
...
+                                     {"acronym": "LBIO4567",
+                                      "is_enrolled": False,
+                                      "etat_to_inscr": None},
```
Comment (htmfilho) [inline exam_enrollment/tests/views/test_exam_enrollment_form.py]: etat -> state

File: exam_enrollment/tests/views/test_exam_enrollment_form.py [hunk 13]
```diff
@@ -54,5 +81,69 @@ def test_form_is_correct(self):
...
+                                     {"acronym": "LDROI1111",
+                                      "is_enrolled": False,
+                                      "etat_to_inscr": None}]
```
Comment (htmfilho) [inline exam_enrollment/tests/views/test_exam_enrollment_form.py]: etat -> state

File: exam_enrollment/views/main.py [hunk 14]
```diff
@@ -50,34 +52,83 @@ def choose_offer(request):
...
+def _build_enrollments_by_learning_unit(request):
+    current_number_session = request.POST['current_number_session']
+    enrollments_by_learn_unit = []
+    is_enrolled_by_acronym = _build_dicts_is_enrolled_by_acronym(current_number_session, request)
+    etat_to_inscr_by_acronym = _build_dicts_etat_to_inscr_by_acronym(request)
+    for acronym, etat_to_inscr in etat_to_inscr_by_acronym.items():
+        enrollments_by_learn_unit.append({
+            'acronym': acronym,
+            'is_enrolled': is_enrolled_by_acronym.get(acronym, False),
+            'etat_to_inscr': None if not etat_to_inscr or etat_to_inscr == 'None' else etat_to_inscr
```
Comment (htmfilho) [inline exam_enrollment/views/main.py]: etat -> state

File: exam_enrollment/views/main.py [hunk 15]
```diff
@@ -50,34 +52,83 @@ def choose_offer(request):
...
+        })
+    return enrollments_by_learn_unit
+
+
+def _build_dicts_etat_to_inscr_by_acronym(request):
```
Comment (htmfilho) [inline exam_enrollment/views/main.py]: etat -> state

**Comment by basmot** [comment]
@htmfilho I fixed the form's action and used a DeprecationWarning with a detailed description of this special `etat_to_inscr` field. Could you merge this?

File: exam_enrollment/locale/en/LC_MESSAGES/django.po (hunk 1 above)
Comment (htmfilho) [inline exam_enrollment/locale/en/LC_MESSAGES/django.po]: Peux-tu corriger ça aussi ?

**Pull request merged**